byon_service/byon_exceptions.py

```diff
@@ -0,0 +1,50 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+
+
+class NoResourcesError(Exception):
+    """Raised when there is no servers left to be acquired"""
+
+    def __init__(self):
+        self.STATUS_CODE = 515
```

Comment:
this init is missing a call to the super class init.
you can see here how we currently define new exceptions:
https://github.com/cloudify-cosmo/cloudify-plugins-common/blob/master/cloudify/exceptions.py